### PR TITLE
Instrument recognizer

### PR DESCRIPTION
**Pull request by AustinShutt**
This pull request adds the following:

   This collab downloads and builds an instrument recognition model utilizing the IRMAS dataset by converting the 3 second audio clip (.wav) to a size (128x259) spectrogram. Then utilizes a Convolutional Neural Network (CNN) to recognize key features between the 11 different classes ['Cello', 'Clarinet', 'Flute', 'Acoustic Guitar', 'Electric Guitar', 'Organ', 'Piano', 'Saxophone', 'Trumpet', 'Violin', 'Voice'].
   
  The model is saved as a .h5 file and imported in to our server to be able to predict uploaded audio. The model appears to be around 55 percent accurate. However the designated test data is in a completely different format than the training data, providing multi-class labels and non-uniform audio length. So I will discuss this more in future work.
  
  The server currently makes a prediction on the uploaded file and returns it's prediction in the basic form of a label posted to the frontend. It only looks at the first three seconds of the audio clip. 
  
  Future Work:
- [ ] Split off a test set from the provided training data instead of utilizing the provided test data since the format is completely different in order to truly test the model accuracy on unseen data.
- [ ] Split the uploaded audio from our site in to multiple 3 second increments for multiple predictions from the model. This will allow us to take the greatest aggregate across all predictions for hopefully a more accurate prediction.
- [ ] Look at increasing the accuracy of our recognition model.
- [ ] Figure out how to modify the sheet music pdf output in the correct key utilizing music21 due to the prediction made by our model.
- [ ] Potentially add a .* to .wav file converter for pre-processing

### CODE REVIEW

File: README.md
Comment (jacobRidenour) [inline README.md:1]: We'll have to add detailed instructions on placing the files in /include/ in the future

File: server/utils.py
Comment (jacobRidenour) [inline server/utils.py:1]: Good call making the lilypond path "just work" on both *nix and Windows

**Review by jacobRidenour — APPROVED** [review]
Excellent work!

**Review by TreytonPerkins — APPROVED** [review]
Looks good!

File: README.md
Comment (AustinShutt) [reply README.md:1]: Ok awesome, got a quick update to the readme added to this PR.
